packages/dds/tree/src/test/shared-tree-core/edit-manager/editManagerScenario.ts

```diff
@@ -241,158 +307,224 @@ export function runUnitTestScenario(
 		/**
 		 * The Ack steps of the scenario
 		 */
-		const acks = steps.filter((s) => s.type === "Ack") as readonly UnitTestAckStep[];
+		const acks = steps.filter(
+			(s) => s.type === "Ack",
+		) as readonly UnitTestAckStepWithIntention[];
 		/**
 		 * Index of the "Ack" step in `acks` that matches the next encountered "Push" step
 		 */
 		let iNextAck = 0;
-		for (const step of steps) {
-			const minimumSequenceNumber = computeMinimumSequenceNumber(
-				step.type === "Push" ? localRef : step.seq,
-			);
-			const type = step.type;
-			switch (type) {
-				case "Push": {
-					let seq = step.seq;
-					if (seq === undefined) {
-						seq =
+		/**
+		 * Process a set of steps that are part of the same bunch.
+		 */
+		const processBunchOfSteps = (bunchOfSteps: UnitTestScenarioStepWithIntention[]) => {
+			assert(bunchOfSteps.length > 0, "Invalid test scenario: empty bunch of steps");
```

Comment:
I'm surprised this doesn't get triggered given the "Process the last bunch, if any." step.
I would recommend either returning instead of asserting, or adding a check before calling `processBunchOfSteps` in the last (i.e., "Process the last bunch, if any.") step.

Reply:
That comment is incorrect - There will also be something in the last bunch. The first message in a bunch will not be processed until the next bunch is identified so there will always be something left to process.
I fixed that comment.